prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -275,50 +275,16 @@ class Hl7Serializer(
                     warnings.add(
                         FieldPrecisionMessage(
                             "ORU_R01",
-                            "Unsupported HL7 message type."
+                            "Unsupported HL7 message type.",
+                            ErrorCode.INVALID_HL7_MSG_TYPE_UNSUPPORTED
                         )
                     )
                     return MessageResult(emptyMap(), errors, warnings)
                 }
             }
         } catch (e: HL7Exception) {
             logger.error("${e.localizedMessage} ${e.stackTraceToString()}")
-            when (e) {
-                is EncodingNotSupportedException ->
-                    // This exception error message is a bit cryptic, so let's provide a better one.
-                    errors.add(
-                        InvalidHL7Message(
-                            "Invalid HL7 message format. Please " +
-                                "refer to the HL7 specification and resubmit.",
-                            ErrorType.INVALID_HL7_MESSAGE_FORMAT.type
-                        )
-                    )
-                else ->
-                    if (e.location?.toString() == "PID-29(0)") {
-                        errors.add(
-                            FieldPrecisionMessage(
-                                e.location.toString(),
-                                "Error parsing HL7 message: ${e.localizedMessage}",
-                                ErrorType.INVALID_HL7_MESSAGE_DATE_VALIDATION.type
-                            )
-                        )
-                    } else {
-                        errors.add(
-                            if (e.location != null) {
-                                FieldPrecisionMessage(
-                                    e.location.toString(),
-                                    "Error parsing HL7 message: ${e.localizedMessage}",
-                                    ErrorType.INVALID_HL7_MESSAGE_VALIDATION.type
-                                )
-                            } else {
-                                InvalidHL7Message(
-                                    "Error parsing HL7 message: ${e.localizedMessage}",
-                                    ErrorType.INVALID_HL7_MESSAGE_VALIDATION.type
-                                )
-                            }
-                        )
-                    }
-            }
+            errors.add(hl7HapiErrorProcessor.getExceptionActionMessage(e, schema))
```

Comment:
Much nicer now, this is where the pay-off for all the other work in this PR comes in. Notice how the old code had the following:

`if (e.location?.toString() == "PID-29(0)") {`

The above approach is not extendable, but now hl7HapiErrorProcessor can neatly determine the appropriate message with correct errorCode based on its new data type mapping approach.